Provider Crashing on v3.2.0

Getting the below panic in our pipeline after updating to provider v3.2.0 w/ terraform v1.0.0. Running in Github actions workflow. Let me know what other logs could be helpful, thanks. I pinned to v3.1.1 for the time being.

Looks like it was probably introduced in #1246

### Terraform Version
1.0.0

### Panic Output
```
Stack trace from the terraform-provider-vault_v3.2.0_x4 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 71437 [running]:
github.com/hashicorp/terraform-provider-vault/vault.rabbitMQSecretBackendRoleRead(0xc000c8e000, {0x10d1280, 0xc000b01d60})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vault/vault/resource_rabbitmq_secret_backend_role.go:175 +0x648
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x10dec2d, {0x134f2e8, 0xc0008165c0}, 0x24, {0x10d1280, 0xc000b01d60})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:346 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0001a21c0, {0x134f2e8, 0xc0008165c0}, 0xc0009a0d00, {0x10d1280, 0xc000b01d60})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:635 +0x35b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00000c3a8, {0x134f2e8, 0xc0008165c0}, 0xc000816640)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/grpc_provider.go:576 +0x534
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00012e880, {0x134f390, 0xc0009bdb60}, 0xc000ab8360)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:553 +0x3b0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x109ac20, 0xc00012e880}, {0x134f390, 0xc0009bdb60}, 0xc000ab8300, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000332c40, {0x13654c0, 0xc00024e480}, 0xc000945440, 0xc0002179e0, 0x1b28eb0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:1279 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc000332c40, {0x13654c0, 0xc00024e480}, 0xc000945440, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:1608 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:923 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:921 +0x294

Error: The terraform-provider-vault_v3.2.0_x4 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Steps to Reproduce
1. `terraform apply`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 3.2.1 (Unreleased)
+BUGS:
+* `resource/rabbitmq_secret_backend_role`: Add nil check to resolve panic when reading RabbitMQ role from Vault
+
 ## 3.2.0 (January 19, 2022)
 BUGS:
 * `resource/aws_secret_backend_role`: Ensure all updated fields are applied ([#1277](https://github.com/hashicorp/terraform-provider-vault/pull/1277))

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,6 @@
 ## 3.2.1 (Unreleased)
 BUGS:
-* `resource/rabbitmq_secret_backend_role`: Add nil check to resolve panic when reading RabbitMQ role from Vault
+* `resource/rabbitmq_secret_backend_role`: Add nil check when reading RabbitMQ role from Vault ([#1312](https://github.com/hashicorp/terraform-provider-vault/pull/1312))
 
 ## 3.2.0 (January 19, 2022)
 BUGS:

--- a/vault/resource_rabbitmq_secret_backend_role.go
+++ b/vault/resource_rabbitmq_secret_backend_role.go
@@ -168,12 +168,19 @@ func rabbitMQSecretBackendRoleRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("backend", strings.Join(pathPieces[:len(pathPieces)-2], "/"))
 	d.Set("name", pathPieces[len(pathPieces)-1])
 
-	if err := d.Set("vhost", flattenRabbitMQSecretBackendRoleVhost(secret.Data["vhosts"].(map[string]interface{}))); err != nil {
-		return fmt.Errorf("Error setting vhosts in state: %w", err)
+	vhosts := secret.Data["vhosts"]
+	vhostTopics := secret.Data["vhost_topics"]
+
+	if vhosts != nil {
+		if err := d.Set("vhost", flattenRabbitMQSecretBackendRoleVhost(secret.Data["vhosts"].(map[string]interface{}))); err != nil {
+			return fmt.Errorf("error setting vhosts in state: %w", err)
+		}
 	}
 
-	if err := d.Set("vhost_topic", flattenRabbitMQSecretBackendRoleVhostTopics(secret.Data["vhost_topics"].(map[string]interface{}))); err != nil {
-		return fmt.Errorf("Error setting vhosts topics in state: %w", err)
+	if vhostTopics != nil {
+		if err := d.Set("vhost_topic", flattenRabbitMQSecretBackendRoleVhostTopics(secret.Data["vhost_topics"].(map[string]interface{}))); err != nil {
+			return fmt.Errorf("Error setting vhosts topics in state: %w", err)
+		}
 	}
 
 	return nil

--- a/vault/resource_rabbitmq_secret_backend_role.go
+++ b/vault/resource_rabbitmq_secret_backend_role.go
@@ -168,18 +168,15 @@ func rabbitMQSecretBackendRoleRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("backend", strings.Join(pathPieces[:len(pathPieces)-2], "/"))
 	d.Set("name", pathPieces[len(pathPieces)-1])
 
-	vhosts := secret.Data["vhosts"]
-	vhostTopics := secret.Data["vhost_topics"]
-
-	if vhosts != nil {
-		if err := d.Set("vhost", flattenRabbitMQSecretBackendRoleVhost(secret.Data["vhosts"].(map[string]interface{}))); err != nil {
+	if vhosts, ok := secret.Data["vhosts"]; ok && vhosts != nil {
+		if err := d.Set("vhost", flattenRabbitMQSecretBackendRoleVhost(vhosts.(map[string]interface{}))); err != nil {
 			return fmt.Errorf("error setting vhosts in state: %w", err)
 		}
 	}
 
-	if vhostTopics != nil {
-		if err := d.Set("vhost_topic", flattenRabbitMQSecretBackendRoleVhostTopics(secret.Data["vhost_topics"].(map[string]interface{}))); err != nil {
-			return fmt.Errorf("Error setting vhosts topics in state: %w", err)
+	if vhostTopics, ok := secret.Data["vhost_topics"]; ok && vhostTopics != nil {
+		if err := d.Set("vhost_topic", flattenRabbitMQSecretBackendRoleVhostTopics(vhostTopics.(map[string]interface{}))); err != nil {
+			return fmt.Errorf("error setting vhosts topics in state: %w", err)
 		}
 	}
 

